homeassistant/components/tuya/climate.py

```diff
@@ -0,0 +1,463 @@
+
+    @property
+    def target_temperature_low(self) -> float:
+        """Return the lower bound target temperature."""
+        if self.__is_celsius():
+            if DPCODE_TEMP_SET not in self.tuya_device.function:
+                return 0
+            temp_value = json.loads(
+                self.tuya_device.function.get(DPCODE_TEMP_SET, {}).values
+            )
+            low_value = (
+                temp_value.get("min", 0) * 1.0 / (10 ** self.get_temp_set_scale())
+            )
+            return low_value
+        if DPCODE_TEMP_SET_F not in self.tuya_device.function:
+            return 0
+        temp_value = json.loads(
+            self.tuya_device.function.get(DPCODE_TEMP_SET_F, {}).values
+        )
+        return temp_value.get("min", 0) * 1.0 / (10 ** self.get_temp_set_scale())
+
+    @property
+    def target_temperature_step(self) -> float:
+        """Return target temperature setp."""
+        if (
+            DPCODE_TEMP_SET not in self.tuya_device.status_range
+            and DPCODE_TEMP_SET_F not in self.tuya_device.status_range
+        ):
+            return 1.0
+        __temp_set_value_range = json.loads(
+            self.tuya_device.status_range.get(
+                DPCODE_TEMP_SET if self.__is_celsius() else DPCODE_TEMP_SET_F
+            ).values
+        )
+        return (
+            __temp_set_value_range.get("step", 0)
+            * 1.0
+            / (10 ** self.get_temp_set_scale())
+        )
+
+    @property
+    def target_humidity(self) -> int:
+        """Return target humidity."""
+        return int(self.tuya_device.status.get(DPCODE_HUMIDITY_SET, 0))
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac mode."""
+        if not self.tuya_device.status.get(DPCODE_SWITCH, False):
+            return HVAC_MODE_OFF
+        if DPCODE_MODE not in self.tuya_device.status:
+            return HVAC_MODE_OFF
+        return TUYA_HVAC_TO_HA[self.tuya_device.status.get(DPCODE_MODE)]
+
+    @property
+    def hvac_modes(self) -> list:
+        """Return hvac modes for select."""
+        if DPCODE_MODE not in self.tuya_device.function:
+            return []
+        modes = json.loads(self.tuya_device.function.get(DPCODE_MODE, {}).values).get(
+            "range"
+        )
+
+        _LOGGER.debug("hvac_modes-> %s", modes)
+        hvac_modes = [HVAC_MODE_OFF]
+        for tuya_mode, ha_mode in TUYA_HVAC_TO_HA.items():
+            if tuya_mode in modes:
+                hvac_modes.append(ha_mode)
+
+        return hvac_modes
+
+    @property
+    def preset_modes(self) -> list:
+        """Return available presets."""
+        if DPCODE_MODE not in self.tuya_device.function:
+            return []
+        modes = json.loads(self.tuya_device.function.get(DPCODE_MODE, {}).values).get(
+            "range"
+        )
+        preset_modes = filter(lambda d: d not in TUYA_HVAC_TO_HA.keys(), modes)
```

Comment:
Please use a list comprehension instead of filter + lambda.
```suggestion
        preset_modes = [d for d in modes if d not in TUYA_HVAC_TO_HA]
```

Reply:
fixed